evaluate/repository.go

```diff
@@ -77,3 +78,56 @@ func Repository(logger *log.Logger, resultPath string, model evalmodel.Model, la
 
 	return repositoryAssessment, problems, nil
 }
+
+// TemporaryRepository creates a temporary repository and initializes a git repo in it.
+func TemporaryRepository(testDataPath string, repositoryPath string) (temporaryRepositoryPath string, err error) {
+	temporaryPath, err := os.MkdirTemp("", "eval-dev-quality")
+	if err != nil {
+		return "", pkgerrors.WithStack(err)
+	}
+
+	dataPath := filepath.Join(testDataPath, repositoryPath)
+	temporaryRepositoryPath = filepath.Join(temporaryPath, filepath.Base(repositoryPath))
```

Comment:
Do you really mean `filepath.Base` or should that be `filepath.Dir`? Just making sure because I made such a mistake as well.

Reply:
I want the last element of `repositoryPath` as it is the "identifier" for the repository inside the temporaryPath. We could also use the full repositoryPath but that seems to much imho.